pycaret/internal/pycaret_experiment/tabular_experiment.py

```diff
@@ -906,17 +911,25 @@ def setup(
             "stratifiedkfold",
             "groupkfold",
             "timeseries",
-            "expanding",
-            "rolling",
-            "sliding",
         ]
-        if not (
-            fold_strategy in possible_fold_strategy
-            or is_sklearn_cv_generator(fold_strategy)
-        ):
-            raise TypeError(
-                f"fold_strategy parameter must be either a scikit-learn compatible CV generator object or one of {', '.join(possible_fold_strategy)}."
-            )
+        possible_time_series_fold_strategies = ["expanding", "sliding", "rolling"]
+
+        if self._ml_usecase != MLUsecase.TIME_SERIES:
+            if not (
+                fold_strategy in possible_fold_strategy
+                or is_sklearn_cv_generator(fold_strategy)
+            ):
+                raise TypeError(
+                    f"fold_strategy parameter must be either a scikit-learn compatible CV generator object or one of {', '.join(possible_fold_strategy)}."
+                )
+        elif self._ml_usecase == MLUsecase.TIME_SERIES:
+            if not (
+                fold_strategy in possible_time_series_fold_strategies
+                or is_sklearn_cv_generator(fold_strategy)
```

Comment:
`is_sklearn_cv_generator` from which module is imported ?

Reply:
It is defined in `pycaret` itself. Just looks to see if there is a split method in the object.